src/zigCodeLensProvider.ts

```diff
@@ -0,0 +1,130 @@
+import * as vscode from "vscode";
+
+/**
+ * CodelensProvider
+ */
+export class CodelensProvider implements vscode.CodeLensProvider {
+  private codeLenses: vscode.CodeLens[] = [];
+  private _onDidChangeCodeLenses: vscode.EventEmitter<void> =
+    new vscode.EventEmitter<void>();
+  public readonly onDidChangeCodeLenses: vscode.Event<void> =
+    this._onDidChangeCodeLenses.event;
+
+  constructor() {
+    vscode.workspace.onDidChangeConfiguration((_) => {
+      this._onDidChangeCodeLenses.fire();
+    });
+  }
+
+  public provideCodeLenses(
+    document: vscode.TextDocument,
+    token: vscode.CancellationToken
+  ): vscode.CodeLens[] | Thenable<vscode.CodeLens[]> {
+    this.codeLenses = [];
+    const text = document.getText();
+
+    var was_newline = false;
+    var test_keyword_start = -1;
+    for (let i = 0; i < text.length; i++) {
+      // test "foo"
+      // ^
+      if (
+        was_newline &&
+        text.length > i + 4 &&
+        text[i] === "t" &&
+        text[i + 1] === "e" &&
+        text[i + 2] === "s" &&
+        text[i + 3] === "t" &&
+        (text[i + 4] === " " || text[i + 4] === "\n")
+      ) {
```

Comment:
This hardly seems like the best solution and breaks easily with badly formatted code:
```zig
test"foo" {}

test
"foo" {}
```